third_party/amd/backend/compiler.py

```diff
@@ -30,12 +31,25 @@ class HIPOptions:
     matrix_inst_shape: int = 0
     max_num_imprecise_acc_default: int = 0
 
+    @staticmethod
+    def get_warp_size(arch: str) -> int:
+        # 64 is not supported for RDNA for now
+        if 'gfx10' in arch or 'gfx11' in arch:
+            return 32
+        if 'gfx9' in arch:
+            return 64
+        print("Warning: Unexpected device. Wave Size is set to 64.")
+        return 64 # Default value
+
     def __post_init__(self):
         default_libdir = Path(__file__).parent / 'lib'
         extern_libs = dict() if self.extern_libs is None else dict(self.extern_libs)
+        if(self.warp_size == 0):
```

Comment:
Why are we checking if warp_size is 0 here? Sorry, I'm not understanding the logic here

Reply:
I'm checking it here to be sure, it was not set up by user. It might be useful when we 're doing some debug or compiling on the not target machine.